### PR TITLE
refactor(gatsby-plugin-netlify-cms) preview styles during development using MiniCssExtractPlugin and HMR

### LINKED ISSUE #15126: [gatsby-plugin-netlify-cms] Preview stylesheet is not registering
## Description

After updating `gatsby-plugin-netlify-cms` to `4.1.0` from `3.0.9`, none of the netlify preview templates have styles applied.

What's interesting is if I import `netlify-cms` instead of `netlify-cms-app`, this problem goes away. Looking at the chrome's network tab, it appears `cms.css` is not getting picked up when importing `netlify-cms-app`

cms.js
```javascript
import CMS from `netlify-cms`;
...
CMS.registerPreviewStyle(`cms.css`);
```

It seems like the only [documented difference](https://www.npmjs.com/package/netlify-cms-app) that probably matters between `netlify-cms` and `netlify-cms-app` is `react` and `react-dom` are now peerDependencies - although this doesn't really explain why this is happening. 

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.
```
System:
    OS: Linux 4.19 Debian GNU/Linux 9 (stretch) 9 (stretch)
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Shell: 4.4.23 -
/nix/store/mcyvp1s45scjkkhyb1m16aqlsb8wr6hs-bash-interactive-4.4-p23/bin/bash
  Binaries:
    Node: 10.15.3 -
/nix/store/47wbs3fgygg6i7za4knbcqpmrwlckqy4-bifrost-env/bin/node
    Yarn: 1.16.0 - /usr/bin/yarn
    npm: 6.4.1 - /nix/store/47wbs3fgygg6i7za4knbcqpmrwlckqy4-bifrost-env/bin/npm
  Languages:
    Python: 3.6.8 - /home/jchi/Projects/bifrost/backend/venv/bin/python
  Browsers:
    Chrome: 73.0.3683.103
    Firefox: 60.7.1
  npmPackages:
    gatsby: 2.10.4 => 2.10.4
    gatsby-cli: 2.7.2 => 2.7.2
    gatsby-link: 2.2.0 => 2.2.0
    gatsby-plugin-intercom-spa: 0.1.0 => 0.1.0
    gatsby-plugin-netlify-cms: 4.1.0-dev-1561486008141 => 4.1.0-dev-1561486008141
    gatsby-plugin-polyfill-io: 1.1.0 => 1.1.0
    gatsby-plugin-react-helmet: 3.1.0 => 3.1.0
    gatsby-plugin-root-import: 2.0.5 => 2.0.5
    gatsby-plugin-sass: 2.1.0 => 2.1.0
    gatsby-plugin-sentry: 1.0.1 => 1.0.1
    gatsby-plugin-sitemap: 2.2.0 => 2.2.0
    gatsby-source-apiserver: 2.1.3 => 2.1.3
    gatsby-source-filesystem: 2.1.0 => 2.1.0
    gatsby-transformer-json: 2.2.0 => 2.2.0
```

### PR DESCRIPTION
**Pull request by entr**
## Description

This aims to provide more straightforward approach to add preview styles to netlify-cms previews during `develop` without workarounds. Basically this reverts some of the changes introduced in #19575 and enables `MiniCssExtractPlugin.loader` with HMR mode for most of the webpack rules that handle css including `sass` and `less`.

## Related Issues

Related to #19575
Fixes #15126 and #20428

### CODE REVIEW

**Comment by erezrokah** [comment]
This looks awesome @entr! Let me do some testing just to be sure

File: packages/gatsby-plugin-netlify-cms/src/gatsby-node.js [hunk 1]
```diff
@@ -29,7 +29,27 @@ function deepMap(obj, fn) {
...
 
-function replaceRule(value) {
+const cssTests = []
+
+function isCssRule({ test }) {
+  return test instanceof RegExp && cssTests.includes(test.toString())
+}
+
+function replaceStyleLoader(rule) {
+  if (rule.loader.includes(`style-loader`)) {
```
Comment (blainekasten) [inline packages/gatsby-plugin-netlify-cms/src/gatsby-node.js]: This function is meant to be used within `replaceRule` after it passes the stage check. I wonder if it makes sense for future developers to put the stage check in this function, or inline this function inside of `replaceRule`. I suppose the potential risk I see is that `replaceStyleLoader` seems generic and reusable, but it's intended to only apply to develop.

Comment (entr) [reply packages/gatsby-plugin-netlify-cms/src/gatsby-node.js]: Thanks for looking into this. Totally makes sense. I've moved it into `replaceRule()`.

**Comment by erezrokah** [comment]
@blainekasten Thank you for merging this so quickly. I completely agree with the testing aspect.
We need to think of a way to keep the plugin aligned with Gatsby build configuration as it changes and improves as this is where we've had issues in the past.
I'll have to think of the best way to do it without affecting Gatsby core development (maybe a separate project that automatically updates to latest Gatsby and checks compatibility with the plugin).

**Pull request merged**